x-pack/plugins/cloud/public/types.ts

```diff
@@ -0,0 +1,111 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import type { FC } from 'react';
+
+export interface CloudStart {
+  /**
+   * A React component that provides a pre-wired `React.Context` which connects components to Cloud services.
+   */
+  CloudContextProvider: FC<{}>;
+  /**
+   * `true` when Kibana is running on Elastic Cloud.
+   */
+  isCloudEnabled: boolean;
+  /**
+   * Cloud ID. Undefined if not running on Cloud.
+   */
+  cloudId?: string;
+  /**
+   * The full URL to the deployment management page on Elastic Cloud. Undefined if not running on Cloud.
+   */
+  deploymentUrl?: string;
+  /**
+   * The full URL to the user profile page on Elastic Cloud. Undefined if not running on Cloud.
+   */
+  profileUrl?: string;
+  /**
+   * The full URL to the organization management page on Elastic Cloud. Undefined if not running on Cloud.
+   */
+  organizationUrl?: string;
+  /**
+   * The full URL to the elasticsearch cluster.
+   */
+  elasticsearchUrl?: string;
+  /**
+   * The full URL to the Kibana deployment.
+   */
+  kibanaUrl?: string;
```

Comment:
Just out of curiosity, is there a difference between this and `server.publicBaseUrl`? (Other than the fact that this will only be available in a cloud environment?)

Reply:
There is. `server.publicBaseUrl` is an arbitrary URL  eventually set by the user, that is supposed to reflect the "public" way of accessing Kibana.

The url decrypted from the cloudId is
1. only available on cloud (obviously)
2. reflecting the "internal" way of accessing the deployment.